zeroex/order_validator.go

```diff
@@ -654,6 +685,108 @@ func (o *OrderValidator) isSupportedAssetData(assetData []byte) bool {
 	return true
 }
 
+// findChunkSize uses a limited-depth binary-search to find a chunk size that is less then the
+// max content length
+func (o *OrderValidator) findChunkSize(signedOrders []*SignedOrder) int64 {
+	// Determine the lowest and highest possible signedOrders limits to binary search in between
+	lo := int64(0)
+	hi := int64(len(signedOrders))
+	min := int64(1)
+
+	// Execute the binary search and hone in on an executable payload size of signedOrders
+	maxBinarySearchDepth := 7
+	depth := 0
+	for lo+1 < hi {
+		depth++
+		mid := (hi + lo) / 2
+		if o.isBelowMaxContentLength(signedOrders[:mid]) {
+			lo = mid
+		} else {
+			hi = mid
+		}
+		if depth == maxBinarySearchDepth {
+			break
+		}
+	}
+
+	if !o.isBelowMaxContentLength(signedOrders[:hi]) {
+		// Reject the signedOrders if it still fails at the lowest allowance
+		if hi == min {
+			// We don't ever expect this to happen since the maxOrderSizeInBytes is much smaller then the
+			// maxRequestContentLength. This error requires a single order to be larger then the
+			// maxRequestContentLength, which would only happen if someone set the maxRequestContentLength
+			// to a tiny number.
+			err := errors.New("A single signedOrder was found that was larger then the maxRequestContentLength. Increase the maxRequestContentLength")
+			log.Panic(err)
+		}
+		hi = lo
+	}
+
+	return hi
+}
+
+// isBelowMaxContentLength checks if the signedOrders once encoded into a payload is
+// below the max content length.
+func (o *OrderValidator) isBelowMaxContentLength(signedOrders []*SignedOrder) bool {
+	orders := []wrappers.OrderWithoutExchangeAddress{}
+	for _, signedOrder := range signedOrders {
+		orders = append(orders, signedOrder.ConvertToOrderWithoutExchangeAddress())
+	}
+	signatures := [][]byte{}
+	for _, signedOrder := range signedOrders {
+		signatures = append(signatures, signedOrder.Signature)
+	}
+
+	payloadSize, err := o.computePayloadSize(getOrderRelevantStatesMethodName, orders, signatures)
+	if err != nil {
+		return false
+	}
+	return payloadSize < o.maxRequestContentLength
+}
+
+func (o *OrderValidator) computePayloadSize(method string, params ...interface{}) (int64, error) {
+	data, err := o.orderValidationUtilsABI.Pack(method, params...)
```

Comment:
This is probably okay for now but some of what we're encoding here is known ahead of time. For example, we know the rest of the JSON RPC payload will always be the same and only the size of the `data` field will change.

Reply:
That's true. I like keeping it because it's more straightforward to refactor if the form/fields of the payload ever change, whereas if we just had a constant number of bytes here, it wouldn't be as clear how it should be changed. Open to getting rid of this though if you feel strongly.

Reply:
Agreed. Replaced with a constant.